### PR TITLE
Enhancement/2426 interactivity pie chart

### PR DESCRIPTION
**Pull request by asvinb**
## Summary

Addresses issue #2426 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js [hunk 1]
```diff
@@ -83,8 +83,15 @@ export default function UserDimensionsPieChart( { dimensionName, entityURL, sour
 					const { dataTable } = GoogleChart.charts.get( 'user-dimensions-pie-chart' ) || {};
 					if ( dataTable ) {
 						const dimensionValue = dataTable.getValue( row, 0 );
-						setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionValue } );
+
+						if ( 'Others' !== dimensionValue ) {
+							setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionValue } );
+						} else {
+							setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionName, dimensionValue: '' } );
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js:90]: @asvinb no need to set the `dimensionName` here and in the next `else` section. Please, remove it and leave only the `dimensionValue` property.

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js [hunk 2]
```diff
@@ -83,8 +83,15 @@ export default function UserDimensionsPieChart( { dimensionName, entityURL, sour
 					const { dataTable } = GoogleChart.charts.get( 'user-dimensions-pie-chart' ) || {};
 					if ( dataTable ) {
 						const dimensionValue = dataTable.getValue( row, 0 );
-						setValues( FORM_ALL_TRAFFIC_WIDGET, { dimensionValue } );
+
+						if ( 'Others' !== dimensionValue ) {
```
Comment (eugene-manuilov) [inline assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js:87]: @asvinb "Others" is translatable, so the `dimensionValue` value will be different when another locale is selected for the site. We need to translate this `Others` as well to make sure it works for all languages.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @asvinb! There are a few things that we need to address before this PR can be approved. Could you please check my comments and address them? Also, once you make changes, could you please change `setValues` to use ternary operator instead of if/else condition?

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js (hunk 1 above)
Comment (asvinb) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js:90]: @eugene-manuilov hmm, it's mentioned in the AC that the dimensionName must be set, unless the AC is outdated?

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js (hunk 2 above)
Comment (asvinb) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js]: Updated @eugene-manuilov

File: assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js (hunk 1 above)
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js:90]: It turned out that it wouldn't work correctly if we set the dimension name here.

Comment (asvinb) [reply assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidgetV2/UserDimensionsPieChart.js]: Gotcha! Updated @eugene-manuilov

**Review by eugene-manuilov — APPROVED** [review]
Awesome! LGTM!

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**